internal/provider/resource_monitor_test.go

```diff
@@ -120,6 +120,108 @@ func TestResourceMonitor(t *testing.T) {
 	})
 }
 
+func TestResourceMonitorWithHeaders(t *testing.T) {
+	var data atomic.Value
+	server := createTestServer(t, &data)
+	defer server.Close()
+
+	var url = "http://example.com"
+	var monitorType = "status"
+
+	resource.Test(t, resource.TestCase{
+		IsUnitTest: true,
+		ProviderFactories: map[string]func() (*schema.Provider, error){
+			"betteruptime": func() (*schema.Provider, error) {
+				return New(WithURL(server.URL)), nil
+			},
+		},
+		Steps: []resource.TestStep{
+			// Step 1 - create.
+			{
+				Config: fmt.Sprintf(`
+				provider "betteruptime" {
+					api_token = "foo"
+				}
+
+				resource "betteruptime_monitor" "this" {
+					url          = "%s"
+					monitor_type = "%s"
+					request_headers = [
+						{
+							name  = "X-TEST"
+							value = "test"
+						}
+					]
+				}
+				`, url, monitorType),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("betteruptime_monitor.this", "id"),
+					resource.TestCheckResourceAttr("betteruptime_monitor.this", "url", url),
+					resource.TestCheckResourceAttr("betteruptime_monitor.this", "monitor_type", monitorType),
+					resource.TestCheckResourceAttrSet("betteruptime_monitor.this", "request_headers.0.id"),
+					resource.TestCheckResourceAttr("betteruptime_monitor.this", "request_headers.0.name", "X-TEST"),
+					resource.TestCheckResourceAttr("betteruptime_monitor.this", "request_headers.0.value", "test"),
+				),
+			},
+			// Step 2 - add another request header.
+			{
+				Config: fmt.Sprintf(`
+				provider "betteruptime" {
+					api_token = "foo"
+				}
+
+				resource "betteruptime_monitor" "this" {
+					url          = "%s"
+					monitor_type = "%s"
+					request_headers = [
+						{
+							name  = "X-TEST"
+							value = "test"
+						},
+						{
+							name  = "X-TEST-2"
+							value = "test-2"
+						}
+					]
+				}
+				`, url, monitorType),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("betteruptime_monitor.this", "request_headers.0.name", "X-TEST"),
+					resource.TestCheckResourceAttr("betteruptime_monitor.this", "request_headers.0.value", "test"),
+					resource.TestCheckResourceAttr("betteruptime_monitor.this", "request_headers.1.name", "X-TEST-2"),
+					resource.TestCheckResourceAttr("betteruptime_monitor.this", "request_headers.1.value", "test-2"),
+					resource.TestCheckNoResourceAttr("betteruptime_monitor.this", "request_headers.2.name"),
```

Comment:
@adikus in this case, will the length of the `request_headers` be 2 or 3? Can we test for that as well, please? (e.g. that there's not an empty map/object at the end of the request_headers array)